babeld/src/ubus.c

```diff
@@ -0,0 +1,373 @@
+#include <stdint.h>
+#include <sys/select.h>
+
+#include <libubox/blob.h>
+#include <libubox/blobmsg.h>
+#include <libubox/list.h>
+
+#include <arpa/inet.h>
+#include <net/if.h>
+#include <netinet/in.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+
+#include <stdlib.h>
+
+#include "babeld.h"
+#include "configuration.h"
+#include "interface.h"
+#include "kernel.h"
+#include "local.h"
+#include "message.h"
+#include "neighbour.h"
+#include "net.h"
+#include "resend.h"
+#include "route.h"
+#include "rule.h"
+#include "source.h"
+#include "util.h"
+#include "version.h"
+#include "xroute.h"
+
+#include "ubus.h"
+
+// config option for ubus bindings
+int ubus_bindings = 0;
+
+// struct definition for sorting xroutes
```

Comment:
Are we really sorting them? It looked to me we just separate v4 and v6.

// List of exported routes (to be used with ubox's list helpers)

idem below